PterodactylLib/Classes/Pterodactyl.swift

```diff
@@ -93,6 +82,39 @@ public class Pterodactyl {
         execute(request: request)
     }
 
+    public func deleteDefaults(for keys: [String]) {
+        guard let endpointUrl = url(for: .deleteDefaults) else { return }
+
+        //Make JSON to send to send to server
+        let json = DeleteDefaultsRequest(
+            simulatorId: ProcessInfo.processInfo.environment["SIMULATOR_UDID"] ?? "booted",
+            appBundleId: targetAppBundleId,
+            keys: keys
+        )
+
+        let encoder = JSONEncoder()
+        guard let data = try? encoder.encode(json) else { return }
+
+        var request = URLRequest(url: endpointUrl)
+        request.httpMethod = "POST"
+        request.httpBody = data
+        request.setValue("application/json", forHTTPHeaderField: "Content-Type")
+        execute(request: request)
+    }
+
+    public func withDefaults(_ defaults: [String: UpdateDefaultsValue], block: (() throws -> Void)) rethrows {
+        updateDefaults(defaults)
+        defer {
+            // Ensure we delete the defaults values even if the block we called throws an error.
+            deleteDefaults(for: Array(defaults.keys))
+        }
+        try block()
+    }
```

Comment:
This is how most code should use the user defaults:
```
pterodactyl.withDefaults(["Test": 100, "monkeys": "banana"]) {
…
}
```
Before executing the block the user defaults will be set and after executing the block, the user defaults will be cleared.